### PR TITLE
ERC20Token-warmup: Add token and safe math lib

### PR DESCRIPTION
**Pull request by jagranata**

### CODE REVIEW

File: ERC20-warmup/jagranata/IC3Token.sol [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  event Transfer(address indexed _from, address indexed _to, uint256 _value);
+  event Approval(address indexed _owner, address indexed _spender, uint256 _value);
+
+  function balanceOf(address _owner) constant returns (uint256 balance) {
+    return balances[_owner];
+  }
+
+  /* deposit 1 wei for 1 ic3 */
+  function deposit () payable returns (bool success) {
+    if (msg.value == 0) return false;
```
Comment (pdaian) [inline ERC20-warmup/jagranata/IC3Token.sol:27]: We may want to allow 0-deposits / return true (this is not specified but worth thinking about)

File: ERC20-warmup/jagranata/IC3Token.sol [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  /* withdraw 1 eth for 1 ic3 */
+  function withdraw (uint256 amount) returns (bool success) {
+    if (amount == 0) return false;
+    if (balances[msg.sender] < amount) return false;
+    balances[msg.sender] = balances[msg.sender].safeSubtract(amount);
+    totalSupply = totalSupply.safeSubtract(amount);
+    bool rv = msg.sender.send(amount);
+    if (!rv) {
```
Comment (pdaian) [inline ERC20-warmup/jagranata/IC3Token.sol:40]: Great!

File: ERC20-warmup/jagranata/IC3Token.sol [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      balances[msg.sender] = balances[msg.sender].safeAdd(amount);
+      totalSupply = totalSupply.safeAdd(amount);
+    }
+    return rv;
+  }
+
+  /* send _value amount of tokens to _to */
+  function transfer (address _to, uint256 _value) returns (bool success) {
+    if (_value == 0) return false;
```
Comment (pdaian) [inline ERC20-warmup/jagranata/IC3Token.sol:49]: We may want to allow this (as above)

**Comment by pdaian** [comment]
Looks great!

File: ERC20-warmup/jagranata/IC3Token.sol (hunk 1 above)
Comment (jagranata) [reply ERC20-warmup/jagranata/IC3Token.sol:27]: :+1:

File: ERC20-warmup/jagranata/IC3Token.sol (hunk 3 above)
Comment (jagranata) [reply ERC20-warmup/jagranata/IC3Token.sol:49]: :+1:

**Pull request merged**